parsers/listParser.go

```diff
@@ -0,0 +1,117 @@
+package parsers
+
+import (
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core/pubkeyConverter"
+	"github.com/multiversx/mx-chain-keys-monitor-go/core"
+)
+
+const commentMarker = "#"
+
+var bech32PubKeyConverter, _ = pubkeyConverter.NewBech32PubkeyConverter(core.AddressLen, core.AddressHRP)
+
+type listParser struct {
+	filename   string
+	mut        sync.RWMutex
+	blsHexKeys []string
+	addresses  []core.Address
+}
+
+// NewListParser creates a new file list parser
+func NewListParser(filename string) *listParser {
+	return &listParser{
+		filename: filename,
+	}
+}
+
+// ParseFile will try to parse to file and split the identities in 2. Errors if something is wrong with the file
+func (parser *listParser) ParseFile() error {
+	dataFile, err := os.ReadFile(parser.filename)
+	if err != nil {
+		return err
+	}
+
+	parser.mut.Lock()
+	defer parser.mut.Unlock()
+
+	parser.emptyLists()
+
+	spitLines := strings.Split(string(dataFile), "\n")
+	for index, line := range spitLines {
+		line = strings.Trim(line, " \t\r\n")
```

Comment:
Does `strings.TrimSpace` help in this context?

Reply:
I guess, will do this in the next PR

Reply:
yup, works, will be done in the next PR